controllers/project.lua

```diff
@@ -298,15 +279,12 @@ ProjectController = {
             -- Parameters:  page, pagesize
             exptime = 60, -- cache expires after 60 seconds
             function (self)
-                local project =
-                    Projects:find(self.params.username, self.params.projectname)
+                local project = assert_project(self, self.params.username, self.params.projectname)
 
-                if not project then yield_error(err.nonexistent_project) end
-                if not project.ispublic then
-                    assert_users_match(self, err.nonexistent_project)
-                end
+                assert_error(can_view_project(self, project), err.nonexistent_project)
```

Comment:
Typically we'd add this to the validations file, but it we don't really need to with this pattern, but it wouldn't hurt.